boot/boot_serial/src/boot_serial.c

```diff
@@ -310,6 +387,130 @@ bs_list(char *buf, int len)
     boot_serial_output();
 }
 
+#ifdef MCUBOOT_SERIAL_IMG_GRP_IMAGE_STATE
+/*
+ * Set image state.
+ */
+static void
+bs_set(char *buf, int len)
+{
+    /*
+     * Expected data format.
+     * {
+     *   "confirm":<true for confirm, false for test>
+     *   "hash":<hash of image (OPTIONAL for single image only)>
+     * }
+     */
+    struct image_header hdr;
+    uint8_t tmpbuf[64];
+    uint32_t area_id;
+    const struct flash_area *fap;
+    uint8_t image_index = 0;
+    struct SetState state_data;
+    size_t decoded_len = 0;
+    uint8_t hash[32];
+    int rc = cbor_decode_SetState((const uint8_t *)buf, len, &state_data, &decoded_len);
+#ifdef MCUBOOT_SERIAL_IMG_GRP_HASH
+    bool found = false;
+#endif
+
+    if ((rc != ZCBOR_SUCCESS) || (len != decoded_len)) {
+        rc = MGMT_ERR_EINVAL;
+        goto out;
+    }
+
+#ifdef MCUBOOT_SERIAL_IMG_GRP_HASH
+    if ((state_data._SetState_hash_present == 1 &&
+         state_data._SetState_hash._SetState_hash.len != sizeof(hash) &&
+         state_data._SetState_hash._SetState_hash.len != 0) ||
+        (state_data._SetState_hash_present == 0 && BOOT_IMAGE_NUMBER > 1)) {
+        /* Hash is required and was not provided or is invalid size */
+        rc = MGMT_ERR_EINVAL;
+        goto out;
+    }
+
+    if (state_data._SetState_hash_present == 1 && state_data._SetState_hash._SetState_hash.len != 0) {
+        for (image_index = 0; image_index < BOOT_IMAGE_NUMBER; ++image_index) {
+            area_id = flash_area_id_from_multi_image_slot(image_index, 1);
+            if (flash_area_open(area_id, &fap)) {
+                continue;
+            }
+
+            rc = BOOT_HOOK_CALL(boot_read_image_header_hook,
```

Comment:
Shouldn't this also be BOOT_HOOK_CALL_FIH?

Reply:
It's a copy and paste from https://github.com/mcu-tools/mcuboot/blob/main/boot/boot_serial/src/boot_serial.c#L249